jobs/jobs_watcher.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package jobs
+
+import (
+	"math/rand"
+	"time"
+
+	l4g "github.com/alecthomas/log4go"
+	"github.com/mattermost/platform/store"
+	"github.com/mattermost/platform/model"
+)
+
+const (
+	WATCHER_POLLING_INTERVAL = 15000
+)
+
+type Watcher struct {
+	store store.Store
+	workers *Workers
+
+	stop    chan bool
+	stopped chan bool
+}
+
+func MakeWatcher(s store.Store, workers *Workers) *Watcher {
+	return &Watcher{
+		store:   s,
+		stop:    make(chan bool, 1),
+		stopped: make(chan bool, 1),
+		workers: workers,
+	}
+}
+
+func (watcher *Watcher) Start() {
+	l4g.Debug("Watcher Started")
+
+	// Delay for some random number of milliseconds before starting to ensure that multiple
+	// instances of the jobserver  don't poll at a time too close to each other.
+	rand.Seed(time.Now().UTC().UnixNano())
+	_ = <-time.After(time.Duration(rand.Intn(WATCHER_POLLING_INTERVAL)) * time.Millisecond)
+
+	running := true
+	for running {
+		select {
+		case <-watcher.stop:
+			l4g.Debug("Watcher: Received stop signal")
+			running = false
+		case <-time.After(15 * time.Second):
```

Comment:
Hmm. Then the constant named `WATCHER_POLLING_INTERVAL` isn't actually the polling interval...